assets/dns/daemonset.yaml

```diff
@@ -2,6 +2,8 @@ kind: DaemonSet
 apiVersion: apps/v1
 # name, namespace and labels are set at runtime
 spec:
+  # minReadySeconds should be 3x the readiness probe's polling interval.
+  minReadySeconds: 9
```

Comment:
The api [docs](https://docs.openshift.com/container-platform/4.12/rest_api/workloads_apis/daemonset-apps-v1.html#spec) say this is The minimum number of seconds for which a newly created DaemonSet pod should be ready without any of its container crashing, for it to be considered available. When it's 0 (default) the pod is immediately ready.  So, does this enforce a delay of 9s?

Reply:
Yes, I believe so.

Reply:
Added a note to the commit message in https://github.com/openshift/cluster-dns-operator/compare/b0ca398b84de1287fe02d22bf9311f0053a02d1c..043594d697605804dae0952b77c741270c0042bc.

Reply:
Does 9 seconds mean that it stays in Pending for 9s until it becomes Ready?  That seems troublesome as a default.

Reply:
The value of `minReadySeconds` is the difference between when a pod is ready and when it is considered "available" (see <https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/daemon-set-v1/#DaemonSetSpec>).  The pod should report "Running" once the container is started, and it should report 2/2 containers ready, same as before.  As far as I can tell, `minReadySeconds` just affects status reporting on the daemonset (viz. `status.numberAvailable` and `status.numberUnavailable`) and behavior of rolling updates.

Reply:
Based on my previous comment, I believe that the note that I added in the commit message in https://github.com/openshift/cluster-dns-operator/compare/b0ca398b84de1287fe02d22bf9311f0053a02d1c..043594d697605804dae0952b77c741270c0042bc is inaccurate, so I have removed it in https://github.com/openshift/cluster-dns-operator/compare/043594d697605804dae0952b77c741270c0042bc..5c4f0ad3430ac1195e990c20398e62bc1681796e.